server/lib/pdf.js

```diff
@@ -28,3 +33,118 @@ export const getTransactionPdf = async (transaction, user) => {
       logger.error(`Error fetching PDF: ${error.message}`);
     });
 };
+
+export const createConsolidatedInvoices = async fromCollective => {
+  const transactions = await models.Transaction.findAll({
+    attributes: ['createdAt', 'HostCollectiveId', 'amountInHostCurrency', 'hostCurrency'],
+    where: {
+      type: 'CREDIT',
+      [Op.or]: [
+        { FromCollectiveId: fromCollective.id, UsingVirtualCardFromCollectiveId: null },
+        { UsingVirtualCardFromCollectiveId: fromCollective.id },
+      ],
+    },
+  });
+
+  const hostsById = {};
+  const invoicesByKey = {};
+  await Promise.map(transactions, async transaction => {
```

Comment:
We should take an opportunity to restructure this with a cleaner/simpler approach.